File: packages/typescript/tests/clabe-vectors.test.ts
import fs from 'fs';
import path from 'path';

import { generateClabe } from '../src/validators/clabe';

type ClabeVector = {
  bank_code: string;
  branch_code: string;
  account_number: string;
  clabe: string;
};

const vectorsPath = path.resolve(__dirname, '../../shared-data/tests/clabe_vectors.json');
const vectors: ClabeVector[] = JSON.parse(fs.readFileSync(vectorsPath, 'utf-8'));

describe('CLABE shared vectors', () => {
  test.each(vectors)('matches %s', (vector) => {
    const clabe = generateClabe(vector.bank_code, vector.branch_code, vector.account_number);
    expect(clabe).toBe(vector.clabe);
  });
});

File: packages/dart/test/clabe_vectors_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:catalogmx/catalogmx.dart';
import 'package:test/test.dart';

void main() {
  test('CLABE shared vectors', () {
    final file = File('../shared-data/tests/clabe_vectors.json');
    final data = jsonDecode(file.readAsStringSync()) as List<dynamic>;

    for (final raw in data) {
      final vector = raw as Map<String, dynamic>;
      final clabe = generateCLABE(
        bankCode: vector['bank_code'] as String,
        branchCode: vector['branch_code'] as String,
        accountNumber: vector['account_number'] as String,
      );
      expect(clabe, vector['clabe']);
    }
  });
}
